app/helpers/provisioning_templates_helper.rb

```diff
@@ -67,6 +67,15 @@ def permitted_actions(template)
     end
   end
 
+  def pxe_with_building_hosts?(template)
+    kinds = ["PXELinux", "PXEGrub"]
+    os_ids = template.operatingsystem_ids
+    template.respond_to?(:template_kind) &&
+      !template.template_kind.nil? &&
+      kinds.include?(template.template_kind.name) &&
+      Host.where(:build => true, :operatingsystem_id => os_ids).count > 0
```

Comment:
If you use `.any?` instead of `.count > 0`, then Rails will optimise the DB call by only fetching any record that matches instead of them all.